homework/index.js

```diff
@@ -1,53 +1,143 @@
 
+  function changeDateFormat(date) {
+    const timeFormat = new Date(date);
+    return timeFormat.toLocaleString();
+  }
+
+  function createTable(table, header, content) {
+    const row = createAndAppend('tr', table);
+    createAndAppend('th', row, { text: header, class: 'header' });
+    createAndAppend('td', row, { text: content, class: 'content' });
+    return row;
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const listItem = createAndAppend('li', ul, { class: 'listItem' });
+    const table = createAndAppend('table', listItem, { class: 'table' });
+    const topRow = createTable(table, 'Repository: ');
+    createAndAppend('a', topRow.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+    });
+    createTable(table, 'Description: ', repo.description);
+    createTable(table, 'Fork count: ', repo.forks);
+    createTable(table, 'Created on: ', changeDateFormat(repo.created_at));
+    createTable(table, 'Updated on: ', changeDateFormat(repo.updated_at));
+  }
+
+  function contributorsSection(url, parent) {
+    fetchJSON(url)
+      .then(repos => {
+        repos.forEach(repo => {
+          const li = createAndAppend('li', parent, { class: 'rightList' });
+          createAndAppend('img', li, {
+            class: 'image',
+            src: repo.avatar_url,
+          });
+          createAndAppend('a', li, {
+            text: repo.login,
+            href: repo.html_url,
+            target: '_blank',
+            class: 'cont-link',
+          });
+          createAndAppend('p', li, {
+            class: 'contributions',
+            text: repo.contributions,
+          });
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', parent, {
+          text: err.message,
+          class: 'alert-error',
+        });
+      });
+  }
+
+  function repoSelect(item, repoItem, parent) {
+    const ul = createAndAppend('ul', parent);
+    const repo = repoItem[item];
+    renderRepoDetails(repo, ul);
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      text: 'HYF Repositories',
+    });
+    const mainElem = createAndAppend('main', root, { class: 'main-container' });
+    const repoContainer = createAndAppend('section', mainElem, {
+      class: 'repo-container',
+    });
+    const ul = createAndAppend('ul', repoContainer);
+    const rightSection = createAndAppend('section', mainElem, {
+      class: 'contributors-container',
+    });
+    const p = createAndAppend('p', rightSection, { class: 'right-title' });
+    p.innerHTML = 'Contributions';
+    const rightUl = createAndAppend('ul', rightSection, { class: 'right-ul' });
+    const select = createAndAppend('select', header, { class: 'test' });
+    fetchJSON(url)
+      .then(repoItem => {
```

Comment:
You are fetching an array of repository objects here, therefore the parameter name should be in plural form: `repoItems`, although I would prefer `repositories` or `repos` for short.